crates/hir/src/lib.rs

```diff
@@ -3232,6 +3234,19 @@ impl Type {
         }
     }
 
+    pub fn array_fields(&self, _db: &dyn HirDatabase) -> Option<(Type, usize)> {
```

Comment:
```suggestion
    pub fn as_array(&self, _db: &dyn HirDatabase) -> Option<(Type, usize)> {
```
naming nit, since arrays don't really have fields